### PR TITLE
Improvement/s3 c 3800 rework failover

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: libV2/tasks/BaseTask.js [hunk 1]
```diff
@@ -104,6 +94,22 @@ class BaseTask extends Process {
     async _join() {
         return this._cache.disconnect();
     }
+
+    withWarp10(func, onError) {
+        return iterIfError(this._warp10Clients, func, error => {
+            const {
+                name, code, message, stack,
+            } = error;
+            logger.error('error during warp 10 request', {
```
Comment (jonathan-gramain) [inline libV2/tasks/BaseTask.js]: Thinking out loud if all errors should be logged in error, or if some errors are "normal" and should be logged in debug/info

File: libV2/utils/func.js [hunk 2]
```diff
@@ -30,7 +30,39 @@ function comprehend(data, func) {
...
+ *
+ * onError, if passed, is called on every error thrown by func;
+ *
+ * @param {Array} items - items to iterate
+ * @param {AsyncFunction} func - function to apply to each item
+ * @param {Function|undefined} onError - optional function called if an error is thrown
+ * @returns {*} -
+ */
+async function iterIfError(items, func, onError) {
+    // console.log(items)
```
Comment (jonathan-gramain) [inline libV2/utils/func.js]: Debugging comments should be removed

**Review by jonathan-gramain — APPROVED** [review]

File: libV2/tasks/BaseTask.js (hunk 1 above)
Comment (tmacro) [reply libV2/tasks/BaseTask.js]: All cases I can think of meet the "error" qualification, though I can see value in being able to silence this. What do you think of changing it to work as a default value for `onError`? That should allow opting out on a case by case basis.

File: libV2/tasks/BaseTask.js [hunk 3]
```diff
@@ -104,6 +94,22 @@ class BaseTask extends Process {
...
+                error: {
+                    name, code, errmsg: message, stack: name !== 'RequestError' ? stack : undefined,
+                },
+            });
+            if (onError) {
+                onError(error);
+            }
```
Comment (tmacro) [inline libV2/tasks/BaseTask.js]:
```suggestion
            if (onError) {
                onError(error);
            } else {
                const {
                    name, code, message, stack,
                } = error;
                logger.error('error during warp 10 request', {
                    error: {
                        name, code, errmsg: message, stack: name !== 'RequestError' ? stack : undefined,
                    },
                });
            }
```

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**